react/LocaleSwitcher.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { useState } from 'react'
+import { useQuery } from 'react-apollo'
+import { useRuntime } from 'vtex.render-runtime'
+import { IconGlobe } from 'vtex.store-icons'
+import { useCssHandles } from 'vtex.css-handles'
+
+import Locales from './queries/locales.gql'
+
+const CSS_HANDLES = [
+  'container',
+  'button',
+  'buttonText',
+  'list',
+  'listElement',
+  'localeIdText',
+]
+
+interface LocalesQuery {
+  languages: {
+    default: string
+    supported: string[]
+  }
+  currentBinding: {
+    supportedLocales: string[]
+  }
+}
+
+interface SupportedLanguage {
+  text: string
+  localeId: string
+}
+
+function getLabel(localeId: string) {
+  return localeId.split('-')[0]
+}
+
+const LocaleSwitcher = () => {
+  const { data, loading, error } = useQuery<LocalesQuery>(Locales)
+  const { culture, emitter } = useRuntime()
+  const [openLocaleSelector, setOpenLocaleSelector] = useState(false)
+
+  const supportedLanguages: string[] =
+    (data &&
+      (data.currentBinding.supportedLocales || data?.languages.supported)) ||
```

Comment:
You're testing if data exists and then you're using the `?` thing (that I don't the name of this feature of js) to get the `.languages.supported`, why not just

```typescript
data?.currentBinding.supportedLocales || data?.languages.supported || []
```
or

```typescript
data && (data.currentBinding.supportedLocales || data.languages.supported)) || []
```
? [just a question mark to finish my question]

Reply:
Nice!